web/profiles/custom/os2loop/modules/os2loop_search_db/src/Helper/Helper.php

```diff
@@ -241,25 +277,85 @@ public function formAlter(array &$form, FormStateInterface $form_state, $form_id
       foreach ($sortLinks as $type => $values) {
         if (empty($parameters)) {
           // Set default active.
-          $form['#sortLinks']['sortDefault']['active'] = TRUE;
+          $variables['sortLinks']['sortDefault']['active'] = TRUE;
         }
         else {
           if (empty(array_diff_assoc($values['requestAlters'], $parameters))) {
-            $form['#sortLinks'][$type]['active'] = TRUE;
+            $variables['sortLinks'][$type]['active'] = TRUE;
           }
           else {
-            $form['#sortLinks'][$type]['active'] = FALSE;
+            $variables['sortLinks'][$type]['active'] = FALSE;
           }
         }
         $newRequest = array_merge($parameters, $values['requestAlters']);
-        $form['#sortLinks'][$type]['url'] = Url::fromRoute('<current>', $newRequest)->toString();
-        $form['#sortLinks'][$type]['label'] = $values['label'];
+        $variables['sortLinks'][$type]['url'] = Url::fromRoute('<current>', $newRequest)->toString();
+        $variables['sortLinks'][$type]['label'] = $values['label'];
       }
     }
   }
 
+  /**
+   * Return the first comment that hits the search string.
+   *
+   * @param \Drupal\node\Entity\Node $node
+   *   A drupal node.
+   * @param string $commentField
+   *   The name of the comment text field on a comment entity.
+   *
+   * @return array|null
+   *   An array containing a comment and text with marked search string.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  public function getSearchedComment(Node $node, string $commentField) {
+    $request = $this->requestStack->getCurrentRequest();
+    $searchString = $request->query->get('search_api_fulltext');
+
+    $cids = $this->entityTypeManager
+      ->getStorage('comment')
+      ->getQuery('AND')
+      ->condition('entity_id', $node->id())
+      ->condition('entity_type', 'node')
+      ->execute();
+    foreach ($cids as $cid) {
+      $comment = $this->commentStorage->load($cid);
+      /** @var \Drupal\comment\CommentInterface $comment */
+      $commentText = strip_tags($comment->get($commentField)->getValue()[0]['value']);
+      if ($searchString) {
+        $hit = stristr($commentText, $searchString);
+        if (FALSE !== $hit) {
+          $hit = $this->truncateText($hit);
+          return [
+            'comment' => $comment,
+            'comment_text' => '...' . str_ireplace($searchString, '<strong>' . $searchString . '</strong>', $hit),
+          ];
+        }
+      }
+
+    }
+    return NULL;
+  }
+
+  /**
+   * Truncate a text and if so add ellipsis.
+   *
+   * @param string $text
+   *   The text.
+   *
+   * @return string
+   *   The truncated text.
+   */
+  private function truncateText($text): string {
+    $newText = substr($text, 0, 250);
+    if (strlen($newText) < strlen($text)) {
+      $newText = $newText . ' ...';
```

Comment:
This can be done in pure CSS, just FYI: https://daily-dev-tips.com/posts/css-modern-multi-line-ellipsis/